pkg/webhook/pod/tikv_deleter.go

```diff
@@ -76,24 +70,29 @@ func (pc *PodAdmissionControl) admitDeleteTiKVPods(payload *admitPayload) *admis
 	}
 
 	var storeInfo *pdapi.StoreInfo
+	var expectedAddress string
 
-	storeId, existed := pod.Labels[label.StoreIDLabelKey]
-	if existed {
-		storeIdInt, err := strconv.Atoi(storeId)
-		if err != nil {
-			klog.Infof("tc[%s/%s]'s tikv pod[%s/%s] failed to delete,%v", namespace, tcName, namespace, name, err)
-			return util.ARFail(err)
-		}
-		// find storeInfo by storeId, if error was not found, deal with it by admitDeleteUselessTiKVPod func
-		storeInfo, err = pdClient.GetStore(uint64(storeIdInt))
-		if err != nil && !strings.HasSuffix(err.Error(), fmt.Sprintf(tikvStoreNotFoundPattern+"\n", storeIdInt)) {
-			klog.Infof("tc[%s/%s]'s tikv pod[%s/%s] failed to delete,%v", namespace, tcName, namespace, name, err)
-			return util.ARFail(err)
+	switch controllerKind {
+	case v1alpha1.TiDBClusterKind:
+		expectedAddress = fmt.Sprintf("%s.%s-tikv-peer.%s.svc:20160", name, controllerName, namespace)
+	case v1alpha1.TiKVGroupKind:
+		expectedAddress = fmt.Sprintf("%s.%s-tikv-group-peer.%s.svc:20160", name, controllerName, namespace)
+	default:
+		err := fmt.Errorf("tikv pod[%s/%s] controlled by unknown controllerKind[%s], forbid to delete", namespace, name, controllerKind)
+		return util.ARFail(err)
```

Comment:
this is unreachable because all possible values of `controllerKind` must be handled in this function
you can add a comment `// unreachable` or just use `panic("unreachable")`